coalib/bearlib/naming_conventions/__init__.py

```diff
@@ -130,3 +130,45 @@ def to_spacecase(string):
     return re.sub('(?<=[^\s])([A-Z])',
                   lambda match: ' ' + match.group(1),
                   string)
+
+
+def to_kebabcase(string):
+    """
+    Converts the given string to kebab-case.
+
+    >>> to_kebabcase('HelloWorld')
+    'hello-world'
+    >>> to_kebabcase('__Init__File__')
+    '--init-file--'
+    >>> to_kebabcase('')
+    ''
+    >>> to_kebabcase('already-kebab-case')
+    'already-kebab-case'
+    >>> to_kebabcase('   string  ')
+    '---string--'
+    >>> to_kebabcase('ABCde.F.G..H..IH')
+    'a-b-cde.f.g..h..i-h'
```

Comment:
IMO a - (hyphen) at the start of the string should not really be there. And since there is no standard for kebab case conversion, we can take this liberty I guess. If we were to parallel kebab case to snake case, In snake case also we have all lower case characters, it's just that the hyphen is replaced with the an underscore.